src/main/java/com/bookstore/repository/ShoppingCartRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.bookstore.repository;
+
+import com.bookstore.entity.ShoppingCart;
+import com.bookstore.entity.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @EntityGraph(attributePaths = {"cartItems", "cartItems.book"})
+    Optional<ShoppingCart> getByUserId(Long userId);
+
+    Optional<ShoppingCart> findByUser(User currentUser);
+
+    Optional<ShoppingCart> findByUserAndIsDeletedFalse(User user);
```

Comment:
You need use `@EntityGraph` or `@Query with JOIN FETCH`  to load `CartItem` collection, otherwise it will cause N + 1 problem in your service